velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -162,6 +162,42 @@ class Subtraction {
   }
 };
 
+class Multiply {
+ public:
+  template <typename R, typename A, typename B>
+  inline static void
+  apply(R& r, const A& a, const B& b, uint8_t aRescale, uint8_t bRescale) {
+    r.setUnscaledValue(checkedMultiply<int128_t>(
+        checkedMultiply<int128_t>(a.unscaledValue(), b.unscaledValue()),
+        DecimalUtil::kPowersOfTen[aRescale + bRescale]));
+  }
+
+  inline static uint8_t
+  computeRescaleFactor(uint8_t fromScale, uint8_t toScale, uint8_t rScale = 0) {
+    return 0;
+  }
+
+  inline static std::pair<uint8_t, uint8_t> computeResultPrecisionScale(
+      const uint8_t aPrecision,
+      const uint8_t aScale,
+      const uint8_t bPrecision,
+      const uint8_t bScale) {
+    return {std::min(38, aPrecision + bPrecision), aScale + bScale};
```

Comment:
How we ensure aScale + bScale doesn't exceed 128? Thanks!

Reply:
These precision and scale values have limits checked while creating the Decimal data type.

Reply:
If the value overflows, the check in type won't catch it, no?

Reply:
@xiaoxmeng I think it's ok since there is a check in type that `scale <= precision <= 38`

Reply:
@karteekmurthys @majetideepak Shall we change the result scale to `min(min(38, aPrecision + bPrecision), aScale + bScale)`?

Reply:
@Yuhta we are returning a pair<> here. Do you mean to say `min(38, aScale + bScale)`?
We can do this in another PR.